python/ctsm/site_and_regional/tower_site.py

```diff
@@ -0,0 +1,413 @@
+                existingcompname = case.get_value("COMPSET")
+                match = re.search("^HIST", existingcompname, flags=re.IGNORECASE)
+                if re.search("^HIST", compset, flags=re.IGNORECASE) is None:
+                    expect(
+                        match is None,
+                        """Existing base case is a historical type and should not be
+                        --rerun with the --overwrite option""",
+                    )
+                else:
+                    expect(
+                        match is not None,
+                        """Existing base case should be a historical type and is not
+                        --rerun with the --overwrite option""",
+                    )
+                # reset the case
+                case.case_setup(reset=True)
+            case_path = case.get_value("CASEROOT")
+
+            if setup_only:
+                return case_path
+
+            print("---- base case build ------")
+            print("--- This may take a while and you may see WARNING messages ---")
+            # always walk through the build process to make sure it's up to date.
+            initial_time = time.time()
+            build.case_build(case_path, case=case)
+            end_time = time.time()
+            total = end_time - initial_time
+            print("Time required to building the base case: {} s.".format(total))
+            # update case_path to be the full path to the base case
+        return case_path
+
+    # pylint: disable=no-self-use
+    def get_batch_query(self, case):
+        """
+        Function for querying the batch queue query command for a case, depending on the
+        user's batch system.
+        Args:
+        case:
+            case object
+        """
+
+        if case.get_value("BATCH_SYSTEM") == "none":
+            return "none"
+        return case.get_value("batch_query")
+
+    def modify_user_nl(self, case_root, run_type, rundir, site_lines=None):
+        """
+        Modify user namelist. If transient, include finidat in user_nl;
+        Otherwise, adjust user_nl to include different mfilt, nhtfrq, and variables in hist_fincl1.
+        """
+        user_nl_fname = os.path.join(case_root, "user_nl_clm")
+        user_nl_lines = None
+        if run_type == "transient":
+            if self.finidat:
+                user_nl_lines = [
+                    "finidat = '{}/inputdata/lnd/ctsm/initdata/{}'".format(rundir, self.finidat)
+                ]
+        else:
+            user_nl_lines = [
+                "hist_fincl2 = ''",
+                "hist_mfilt = 20",
+                "hist_nhtfrq = -8760",
+                "hist_empty_htapes = .true.",
+            ] + site_lines
+
+        if user_nl_lines:
+            with open(user_nl_fname, "a") as nl_file:
+                for line in user_nl_lines:
+                    nl_file.write("{}\n".format(line))
+
+    def set_ref_case(self, case):
```

Comment:
This could be unit tested.

Reply:
I'll point you to a way to do this with a dummy case as a way to do this in a unit test.

And we will do this at a later PR.

Reply:
#2413 talks about refactoring to make things unit-testable.